src/test/java/com/marklogic/kafka/connect/source/ConstraintStorageTest.java

```diff
@@ -0,0 +1,110 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.junit5.PermissionsTester;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+class ConstraintStorageTest extends AbstractIntegrationSourceTest {
```

Comment:
I think these two tests can be merged, as there's really just one scenario:

1. Load the fifteen authors
2. Run the Task
3. Verify that the constraint value document was written correctly (as you're doing already)
4. Load 3 more authors - two with IDs higher than the highest one that was captured, and one with an ID that is lowest that the highest one
5. Run the Task again
6. Verify that only the two new authors with a higher ID are returned
7. And then verify that the constraint value doc was updated as well

Reply:
Also, I think it's worth adding an error scenario too - i.e. do something to cause the read of the JSON doc to fail, such as first writing some random XML to that URI. Then load the 15 authors, and run the Task, and verify that you get null back since the read of the JSON doc failed.